### PR TITLE
fix: use fork instead of spawn on npminstall mode

### PR DESCRIPTION
**Pull request by fengmk2**
can fix install cnpm itself fail on Windows

### CODE REVIEW

**Comment by fengmk2** [comment]
在 Windows 上测试成功

image

File: origin_npm.js [hunk 1]
```diff
@@ -52,11 +54,21 @@ if (program.proxy) {
...
+  process.stdout,
+  process.stderr,
+];
 
 if (isInstall) {
   npmBin = path.join(__dirname, 'node_modules', '.bin', installer);
   if (installer === 'npminstall') {
+    // use fork to spawn can fix install cnpm itself fail on Windows
+    execMethod = cp.fork;
+    stdio.push('ipc');
```
Comment (fengmk2) [inline origin_npm.js]: 这个 ipc stdio 参数比较特殊，看文档有说明 https://npm.taobao.org/mirrors/node/latest/docs/api/child_process.html#child_process_child_process_fork_modulepath_args_options

> stdio <Array> Supports the array version of child_process.spawn()'s stdio option. When this option is provided, it overrides silent. The array must contain exactly one item with value 'ipc' or an error will be thrown. For instance [0, 1, 2, 'ipc'].

**Comment by dead-horse** [comment]
+1

**Comment by dead-horse** [comment]
4.4.2

**Pull request merged**